### PR TITLE
WMS temporal support docs

### PR DESCRIPTION
**Pull request by vikineema**
Documenting the temporal support for WMS-T layers in QGIS.

### CODE REVIEW

File: web_services/index.ipynb [hunk 1]
```diff
@@ -105,6 +105,163 @@
...
+   "source": [
+    "**How to add WMS-T layers using QGIS** "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "1. Connect to the Web Map Service in QGIS by following the instructions in [Setting up WMS tutorial](#Tutorial:-Setting-up-WMS).\n",
+    "2. In the QGIS **Browser Panel** navigate to the **WMS/WMTS** location and expand it to show the **DE Africa Services** resource. \n",
```
Comment (Samweli) [inline web_services/index.ipynb:127]:
```suggestion
    "2. In the QGIS **Browser Panel** navigate to the **WMS/WMTS** provider item and expand it to show the **DE Africa Services** connection. \n",
```

File: web_services/index.ipynb [hunk 2]
```diff
@@ -105,6 +105,163 @@
...
+    "\n",
+    "<img align=\"middle\" src=\"../_static/web_services/ows_tutorial_7.png\" alt=\"QGIS - WMS-T layers\" width=\"300\">\n",
+    "\n",
+    "3. Select a WMS-T layer from the **DE Africa Services** resource by either double clicking on the layer in the **Browser Panel** or right clicking on it and selecting **Add Layer to Project** to add the layer onto the map canvas.\n",
```
Comment (Samweli) [inline web_services/index.ipynb:131]:
```suggestion
    "3. Select a WMS-T layer from the **DE Africa Services** connection layers by either double clicking on the layer item in the **Browser Panel** or right clicking on it and selecting **Add Layer to Project** to add the layer onto the map canvas.\n",
```

File: web_services/index.ipynb [hunk 3]
```diff
@@ -105,6 +105,163 @@
...
+    "\n",
+    "<img align=\"middle\" src=\"../_static/web_services/ows_tutorial_8.png\" alt=\"QGIS - WMS-T layers\" width=\"300\">\n",
+    "\n",
+    "4. The name of the loaded WMS-T layer is displayed in the **Layers Panel** with a clock symbol ![temporal indicator](../_static/web_services/mIndicatorTemporal.svg) next it to indicate that it is a temporal layer. \n",
```
Comment (Samweli) [inline web_services/index.ipynb:135]:
```suggestion
    "4. The layer will be loaded in the **Layers Panel** with a clock symbol ![temporal indicator](../_static/web_services/mIndicatorTemporal.svg) next it to indicate that it is a temporal layer. \n",
```

File: web_services/index.ipynb [hunk 4]
```diff
@@ -105,6 +105,163 @@
...
+   "metadata": {},
+   "source": [
+    "**How to use the QGIS Temporal Controller with WMS-T layers**"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "1. To use the **Temporal Controller** for WMS-T layers, ensure that in the Temporal section of the layer's **Layer Properties**, the **Dynamic Temporal Control** is selected. You can select the time range to be **Automatic** or **Fixed time range**. \n",
```
Comment (Samweli) [inline web_services/index.ipynb:151]: The paragraphs breaks are not working on the final page, can you take a look and fix this.

File: web_services/index.ipynb [hunk 5]
```diff
@@ -105,6 +105,163 @@
...
+    "For the **Automatic** setting, the temporal handling is delegated to the data provider and time based filtering is applied to the features of the layer such that the features will be visualized for the animation range set for the **Animated temporal navigation**, or the range set for the **Fixed range temporal navigation**. \n",
+    "For the **Fixed time range** setting the user manually sets the temporal range in which all the features of the layer will be visualised only if the animation range, for the Animated temporal navigation, or the Range, for the Fixed range temporal navigation, is within this temporal range. \n",
+    "\n",
+    "<img align=\"middle\" src=\"../_static/web_services/ows_tutorial_10.png\" alt=\"QGIS - WMS-T layers\" width=\"500\">\n",
+    "\n",
+    "<img align=\"middle\" src=\"../_static/web_services/ows_tutorial_11.png\" alt=\"QGIS - WMS-T layers\" width=\"500\">\n",
+    "\n",
+    "2. On the **Map Navigation Toolbar** select the clock symbol for the **Temporal Controller Panel** to activate the **Temporal Controller Panel**. \n",
```
Comment (Samweli) [inline web_services/index.ipynb:159]:
```suggestion
    "2. On the **Map Navigation Toolbar** select the clock symbol for the **Temporal Controller Panel** to activate the **Temporal Controller ** dock widget. \n",
```

File: web_services/index.ipynb [hunk 6]
```diff
@@ -105,6 +105,163 @@
...
+    "\n",
+    "<img align=\"middle\" src=\"../_static/web_services/ows_tutorial_14.png\" alt=\"QGIS - WMS-T layers\" width=\"500\">\n",
+    "\n",
+    "4. To turn on the Animated temporal navigation, select the **Animated temporal navigation** icon ![Temporal Navigation Animated](../_static/web_services/mTemporalNavigationAnimated.svg) on the **Temporal Controller Panel**, and set animation range and the time step for the animation.  \n",
+    "\n",
+    "<img align=\"middle\" src=\"../_static/web_services/ows_tutorial_15.png\" alt=\"QGIS - WMS-T layers\" width=\"500\">\n",
+    "\n",
+    "<img align=\"middle\" src=\"../_static/web_services/ows_tutorial_16.png\" alt=\"QGIS - WMS-T layers\" width=\"500\">\n",
+    "\n",
+    "5. Click on the play icon ![play](../_static/web_services/forward.svg) to view the animation. \n",
```
Comment (Samweli) [inline web_services/index.ipynb:175]: We need to explain also about the use of other navigation buttons apart for the play button.

File: web_services/index.ipynb [hunk 7]
```diff
@@ -105,6 +105,163 @@
...
+    "    * The width and height of the images. \n",
+    "    * The temporal range and time step for the animation.\n",
+    "    \n",
+    "<img align=\"middle\" src=\"../_static/web_services/ows_tutorial_19.png\" alt=\"QGIS - WMS-T layers\" width=\"300\">\n",
+    "\n",
+    "3. The exported images will be saved in the specified ouput directory. \n",
+    "\n",
+    "<img align=\"middle\" src=\"../_static/web_services/ows_tutorial_20.png\" alt=\"QGIS - WMS-T layers\" width=\"500\">\n",
+    "\n",
+    "4. Using your preferred software you can use the images to create an animation."
```
Comment (Samweli) [inline web_services/index.ipynb:208]: This can be a comment on the list item 3 and not a list item itself.

File: web_services/index.ipynb [hunk 8]
```diff
@@ -105,6 +105,163 @@
...
+   "metadata": {},
+   "source": [
+    "**How to use the WMS-T layers in QGIS without the Temporal Controller** "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "1. To use the WMS-T layers in QGIS without the Temporal Controller, first disable the **Dynamic Temporal Control** in the layer's **Layer Properties** to enable the **Static WMS-T Temporal Range**. The default setting is the **Server default** temporal range for visualizing the layer features. This is a predefined temporal range set by the data service provider. You will notice that the layer no longer has the temporal indicator symbol ![temporal indicator](../_static/web_services/mIndicatorTemporal.svg) next to the layer name in the **Layers Panel**.\n",
```
Comment (Samweli) [inline web_services/index.ipynb:222]:
```suggestion
    "1. To use the WMS-T layers in QGIS without the Temporal Controller, first disable the **Dynamic Temporal Control** in the layer's **Layer Properties** to enable the **Static WMS-T Temporal Range** settings. The default setting is the **Server default** temporal range for visualizing the layer features. This is a predefined temporal range set by the data service provider.\n
 You will notice after disabling **Dynamic Temporal Control** that the layer no longer has the temporal indicator symbol ![temporal indicator](../_static/web_services/mIndicatorTemporal.svg) next to the layer name in the **Layers Panel**, this means that the layer is no longer considered a temporal layer by QGIS, hence can't be used by the ** Temporal Controller**\n",
```

also check if the paragraph breaks are working here

File: web_services/index.ipynb [hunk 9]
```diff
@@ -105,6 +105,163 @@
...
+    "\n",
+    "<img align=\"middle\" src=\"../_static/web_services/ows_tutorial_21.png\" alt=\"QGIS - WMS-T layers\" width=\"300\">\n",
+    "\n",
+    "<img align=\"middle\" src=\"../_static/web_services/ows_tutorial_22.png\" alt=\"QGIS - WMS-T layers\" width=\"500\">\n",
+    "\n",
+    "2. Using the **Predefined range** you can set the temporal range for the layer features to be visualized using the **Start date** and **End date**. A time-based filter will be applied to the layer and the only features to be visualized will be features within the predefined temporal range.\n",
```
Comment (Samweli) [inline web_services/index.ipynb:228]:
```suggestion
    "2. Using the **Predefined range** you can set the temporal range for the layer features to be visualized using the **Start date** and **End date**. A time-based filter will be applied to the layer and the fetched features will be the features within the predefined temporal range.\n",
```

File: web_services/index.ipynb [hunk 10]
```diff
@@ -105,6 +105,163 @@
...
+    "\n",
+    "<img align=\"middle\" src=\"../_static/web_services/ows_tutorial_23.png\" alt=\"QGIS - WMS-T layers\" width=\"300\">\n",
+    "\n",
+    "3. To use the **Follow project's temporal range** setting, you must first define the **Start date** and **End date** in the **Temporal** section (Project > Properties> Temporal) of the **Project Properties** . A time-based filter will be applied to the layer and the only features to be visualized will be features within the defined project temporal range.\n",
```
Comment (Samweli) [inline web_services/index.ipynb:232]:
```suggestion
    "3. To use the **Follow project's temporal range** setting, project temporal range in the **Temporal** section (Project > Properties> Temporal) of the **Project Properties** must be set. After setting the project temporal range, applying this setting will result in fetching the WMS layer features that are within the projects temporal range.\n",
```

File: web_services/index.ipynb [hunk 11]
```diff
@@ -105,6 +105,163 @@
...
+   "metadata": {},
+   "source": [
+    "**WMS-T Settings**"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You can also specify the time slice mode used for both the **Dynamic Temporal Control** and the **Static WMS-T Temporal Range**. There are four options available for the Time slice mode: \n",
```
Comment (Samweli) [inline web_services/index.ipynb:252]:
```suggestion
    "You can also specify the time slice mode used for both the **Dynamic Temporal Control** and the **Static WMS-T Temporal Range**. There are four options available for the time slice mode: \n",
```

File: web_services/index.ipynb [hunk 12]
```diff
@@ -105,6 +105,163 @@
...
+    "\n",
+    "* **Use Whole Temporal Range**: Allows the user to set the temporal range for the layer features to be visualized/plotted. \n",
+    "* **Match to Start of Range**: The layer features visualized will be features that match the start of the layer's time range. \n",
+    "* **Match to End of Range**: The layer features visualized will be features that match the end of the layer's time range.\n",
+    "* **Closest Match to Start of Range**: Visualizes the layer features whose date and time stamp are the closest to the start of the defined temporal range. \n",
+    "* **Closet Match to End of Range**: Visualizes the layer features whose date and time stamp are the closest to the end of the defined temporal range. \n",
+    "\n",
+    "You can also enable the option for **Ignoring time compononents** which allows QGIS to only consider the dates when applying a time based filter to the layer features based on a defined temporal range, without considering the time component.\n",
```
Comment (Samweli) [inline web_services/index.ipynb:260]:
```suggestion
    "You can also enable the option for **Ignoring time components** which allows QGIS to only consider the dates when applying a time based filter to the layer features based on a defined temporal range, without considering the time component.\n",
```

**Review by Samweli — CHANGES_REQUESTED** [review]

File: web_services/index.ipynb (hunk 4 above)
Comment (vikineema) [reply web_services/index.ipynb]: Fixed the paragraph breaks. https://github.com/digitalearthafrica/deafrica-docs/pull/41/commits/60cb4691b9ebec77f87459e5729a0a706dc9fd1c

File: web_services/index.ipynb (hunk 7 above)
Comment (vikineema) [reply web_services/index.ipynb]: Fixed this and added an example animation:  https://github.com/digitalearthafrica/deafrica-docs/pull/41/commits/24091f6cd89718efbc7d15423f7852c8a691390d

File: web_services/index.ipynb [hunk 13]
```diff
@@ -105,6 +105,186 @@
...
+    "\n",
+    "    For the **Automatic** setting, the temporal handling is delegated to the data provider and time based filtering is applied to the features of the layer such that the features will be visualized for the animation range set for the **Animated temporal navigation**, or the range set for the **Fixed range temporal navigation**. \n",
+    "    \n",
+    "    For the **Fixed time range** setting the user manually sets the temporal range in which all the features of the layer will be visualised only if the animation range, for the Animated temporal navigation, or the Range, for the Fixed range temporal navigation, is within this temporal range. \n",
+    "\n",
+    "<img align=\"middle\" src=\"../_static/web_services/ows_tutorial_10.png\" alt=\"QGIS - WMS-T layers\" width=\"500\">\n",
+    "\n",
+    "<img align=\"middle\" src=\"../_static/web_services/ows_tutorial_11.png\" alt=\"QGIS - WMS-T layers\" width=\"500\">\n",
+    "\n",
+    "2. On the **Map Navigation Toolbar** select the clock symbol for the **Temporal Controller Panel** to activate the **Temporal Controller ** dock widget. \n",
```
Comment (Samweli) [inline web_services/index.ipynb:161]:
```suggestion
    "2. On the **Map Navigation Toolbar** select the clock symbol for the **Temporal Controller Panel** to activate the **Temporal Controller** dock widget. \n",
```

File: web_services/index.ipynb [hunk 14]
```diff
@@ -105,6 +105,186 @@
...
+    "**How to use the QGIS Temporal Controller with WMS-T layers**"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "1. To use the **Temporal Controller** for WMS-T layers, ensure that in the Temporal section of the layer's **Layer Properties**, the **Dynamic Temporal Control** is selected. You can select the time range to be **Automatic** or **Fixed time range**.\n",
+    "\n",
+    "    For the **Automatic** setting, the temporal handling is delegated to the data provider and time based filtering is applied to the features of the layer such that the features will be visualized for the animation range set for the **Animated temporal navigation**, or the range set for the **Fixed range temporal navigation**. \n",
```
Comment (Samweli) [inline web_services/index.ipynb:153]:
```suggestion
    "    For the **Automatic** setting, the temporal handling is delegated to the data provider and time based filtering is applied to the features of the layer such that the features will be visualized for the animation range set for the **Animated temporal navigation**, or the range set for the **Fixed range temporal navigation**. \n",
    "    \n",
    "<img align=\"middle\" src=\"../_static/web_services/ows_tutorial_10.png\" alt=\"QGIS - WMS-T layers\" width=\"500\">\n",
    "\n",
    "    For the **Fixed time range** setting the user manually sets the temporal range in which all the features of the layer will be visualised only if the animation range, for the Animated temporal navigation, or the Range, for the Fixed range temporal navigation, is within this temporal range. \n",
    "\n",
    "<img align=\"middle\" src=\"../_static/web_services/ows_tutorial_11.png\" alt=\"QGIS - WMS-T layers\" width=\"500\">\n",
    "\n",
```

File: web_services/index.ipynb [hunk 15]
```diff
@@ -105,6 +105,186 @@
...
+    "\n",
+    "<img align=\"middle\" src=\"../_static/web_services/ows_tutorial_10.png\" alt=\"QGIS - WMS-T layers\" width=\"500\">\n",
+    "\n",
+    "<img align=\"middle\" src=\"../_static/web_services/ows_tutorial_11.png\" alt=\"QGIS - WMS-T layers\" width=\"500\">\n",
+    "\n",
+    "2. On the **Map Navigation Toolbar** select the clock symbol for the **Temporal Controller Panel** to activate the **Temporal Controller ** dock widget. \n",
+    "\n",
+    "<img align=\"middle\" src=\"../_static/web_services/ows_tutorial_12.png\" alt=\"QGIS - WMS-T layers\" width=\"500\">\n",
+    "\n",
+    "<img align=\"middle\" src=\"../_static/web_services/ows_tutorial_13.png\" alt=\"QGIS - WMS-T layers\" width=\"500\">\n",
```
Comment (Samweli) [inline web_services/index.ipynb:165]:
```suggestion
    " **Temporal controller** dock widget activated after click on the **Temporal Controller** toolbar icon."
    "<img align=\"middle\" src=\"../_static/web_services/ows_tutorial_13.png\" alt=\"QGIS - WMS-T layers\" width=\"500\">\n",
```

**Review by Samweli — CHANGES_REQUESTED** [review]
Victoria almost there!
Can you edit all of the images `alt `property with relevant comment, also it will be good for the images to contain a title explaining to user what the image is about.

File: web_services/index.ipynb [hunk 16]
```diff
@@ -105,6 +105,199 @@
...
+    "**How to use the QGIS Temporal Controller with WMS-T layers**"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "1. To use the **Temporal Controller** for WMS-T layers, ensure that in the Temporal section of the layer's **Layer Properties**, the **Dynamic Temporal Control** is selected. You can select the time range to be **Automatic** or **Fixed time range**.\n",
+    "\n",
+    "    For the **Automatic** setting, the temporal handling is delegated to the data provider and time based filtering is applied to the features of the layer such that the features will be visualized for the animation range set for the **Animated temporal navigation**, or the range set for the **Fixed range temporal navigation**.\n",
```
Comment (Samweli) [inline web_services/index.ipynb:153]:
```suggestion
    "    For the **Automatic** setting, the temporal handling is delegated to the data provider and time based filtering is applied to the features of the layer such that the features will be visualized for the animation range set by the **Temporal Controller**.\n",
```

File: web_services/index.ipynb [hunk 17]
```diff
@@ -105,6 +105,199 @@
...
+    "    \n",
+    "    <img align=\"middle\" src=\"../_static/web_services/ows_tutorial_10.png\" alt=\"QGIS - Dynamic Temporal Control - Automatic\" width=\"500\">\n",
+    "    \n",
+    "    For the **Fixed time range** setting the user manually sets the temporal range in which all the features of the layer will be visualised only if the animation range, for the Animated temporal navigation, or the Range, for the Fixed range temporal navigation, is within this temporal range.\n",
```
Comment (Samweli) [inline web_services/index.ipynb:157]:
```suggestion
    "    For the **Fixed time range** setting the user manually sets the temporal range in which all the features of the layer will be visualised only if the animation range from the **Temporal Controller**  is within this temporal range.\n",
```

**Review by Samweli — COMMENTED** [review]
@vikineema a few last changes and then we can change the PR status to be ready for merge.

File: web_services/index.ipynb [hunk 18]
```diff
@@ -105,6 +105,199 @@
...
+   "metadata": {},
+   "source": [
+    "**How to use the QGIS Temporal Controller with WMS-T layers**"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "1. To use the **Temporal Controller** for WMS-T layers, ensure that in the Temporal section of the layer's **Layer Properties**, the **Dynamic Temporal Control** is selected. You can select the time range to be **Automatic** or **Fixed time range**.\n",
```
Comment (Samweli) [inline web_services/index.ipynb:151]: Lets make this a paragraph instead of a list item, so the next list item no. 2 will be list item no. 1

**Review by Samweli — APPROVED** [review]

**Review by alexgleith — APPROVED** [review]

**Comment by eefaye** [comment]
Looks great to me. It's all yours to merge @vikineema

**Review by eefaye — APPROVED** [review]

**Pull request merged**